tests/common/devices.py

```diff
@@ -44,6 +44,7 @@ def __init__(self, ansible_adhoc, hostname, connection=None, become_user=None):
                     self.host = ansible_adhoc(become=True, connection=connection)[hostname]
                 else:
                     self.host = ansible_adhoc(become=True, connection=connection, become_user=become_user)[hostname]
+            self.ip = self.host.options["inventory_manager"].get_host(hostname).vars["ansible_host"]
```

Comment:
Do we need to set a default ip for localhost to avoid AttributeError?

Reply:
Updated.